app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/RestaurantListActivityTest.kt

```diff
@@ -0,0 +1,140 @@
+package com.github.sdp.ratemyepfl.activity
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.platform.app.InstrumentationRegistry
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.activity.restaurants.RestaurantListActivity
+import com.github.sdp.ratemyepfl.utils.closeSearchView
+import com.github.sdp.ratemyepfl.utils.openSearchView
+import com.github.sdp.ratemyepfl.utils.typeQuery
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.hamcrest.CoreMatchers
+import org.junit.Rule
+import org.junit.Test
+
+@HiltAndroidTest
+class RestaurantListActivityTest {
+
+    @get:Rule(order = 0)
+    val hiltRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val testRule = ActivityScenarioRule(RestaurantListActivity::class.java)
+
+    @Test
+    fun isRecyclerViewVisibleOnActivityLaunch() {
+        Espresso.onView(ViewMatchers.withId(R.id.reviewableRecyclerView))
```

Comment:
You could rewrite these tests as `onView(withId(...))` for instance to make them more comfortable to read, and apply that to the other tests as well

Reply:
Yes, my bad, Mathias will change it in his refactoring !